### ISSUE TITLE
Support TimeType versus Float64 for time fields

### ISSUE THREAD

**Issue by milktrader** [BenLauwens/SimJulia.jl#22]
I'm working on this on a branch but if this is a "will not happen" issue I'm happy enough to fork it and see where it goes.

**Comment by BenLauwens** [comment]
Hi

Interesting idea! I am rewriting the code and I will experiment with TimeType. I will comment in a couple of days but it seems more natural that time is a TimeType;)

**Comment by milktrader** [comment]
Ok cool great you're open to the idea!

**Comment by BenLauwens** [comment]
In the master branch (very little functionality for the moment;) I implemented the scheduling with TimeType. I am still struggling to restore the previous behavior with Float64. I will try to do this with a SimulationTime type.
Can you verify that this is what you like to use? Thanks!

**Comment by milktrader** [comment]
Will take a look. iirc it was the scheduling methods that were tripping up a trivial `s/Float64/TimeType` approach.

**Comment by milktrader** [comment]
After pulling afd4d3301eef698c2c843f42b17ee0cb4887f61b

````julia
julia> using SimJulia
INFO: Precompiling module SimJulia...
ERROR: LoadError: LoadError: syntax: invalid "::" syntax
 in include at ./boot.jl:261
 in include_from_node1 at ./loading.jl:304
 in include at ./boot.jl:261
 in include_from_node1 at ./loading.jl:304
 [inlined code] from none:2
 in anonymous at no file:0
 in process_options at ./client.jl:257
 in _start at ./client.jl:378
while loading /Users/Administrator/.julia/v0.4/SimJulia/src/base.jl, in expression starting on line 69
while loading /Users/Administrator/.julia/v0.4/SimJulia/src/SimJulia.jl, in expression starting on line 19
ERROR: Failed to precompile SimJulia to /Users/Administrator/.julia/lib/v0.4/SimJulia.ji
 in error at ./error.jl:21
 in compilecache at loading.jl:384
 in require at ./loading.jl:250

julia> versioninfo()
Julia Version 0.4.1
Commit cbe1bee* (2015-11-08 10:33 UTC)
Platform Info:
  System: Darwin (x86_64-apple-darwin13.4.0)
  CPU: Intel(R) Core(TM)2 Duo CPU     P7350  @ 2.00GHz
  WORD_SIZE: 64
  BLAS: libopenblas (USE64BITINT DYNAMIC_ARCH NO_AFFINITY Penryn)
  LAPACK: libopenblas64_
  LIBM: libopenlibm
  LLVM: libLLVM-3.3
````

**Comment by BenLauwens** [comment]
Try with julia v0.5. I did some experiments with the return type feature ...

**Comment by milktrader** [comment]
Yep, that fixes it.

**Comment by BenLauwens** [comment]
I will close this issue. As soon as Julia v0.5 is released I will release SimJulia v0.4 including your request for TimeType time fields. I am still finalising the API (reducing the number of base functions: yield, schedule, append_callback, ... and moving other functions to a utils sub-package).

**Comment by milktrader** [comment]
Sounds good
